pkg/sync/filepath_sync_test.go

```diff
@@ -0,0 +1,73 @@
+package sync_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/open-feature/flagd/pkg/sync"
+	log "github.com/sirupsen/logrus"
+	"os"
+	"testing"
+	"time"
+)
+
+const dirName = "test"
+
+func TestFilePathSync_Notify(t *testing.T) {
+	defer t.Cleanup(cleanup)
+	setup(t)
+
+	fpSync := sync.FilePathSync{URI: dirName}
+
+	t.Run("Create", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		inotifyChan := make(chan sync.INotify)
+		go func() {
+			fpSync.Notify(ctx, inotifyChan)
+		}()
+
+		// enough time for Notify to be ready to receive file watcher events.
+		// time.Sleep in tests is bad practice. A rework to avoid this would be to pass a ready channel to Notify
+		// to read from to determine readiness.
+		time.Sleep(10 * time.Millisecond)
```

Comment:
I've put this `time.Sleep` in place for the reason described by the comment above. I'm against this kind of non-deterministic approach, I'd prefer to rework the `Notify` func to accept a `ready chan <- struct{}` for us to determine when the notifier is at a ready state. Where do you guys stand on this? @toddbaert @james-milligan @beeme1mr

Reply:
I think this is 100% going to be worthwhile to introduce the ```ready chan <- struct{}```, especially if we intend to implement the ```PROVIDER_NOT_READY``` error code as we will need to evaluate this in the check.

In terms of the ```time.Sleep()``` - I find it worrying that there may be a race condition in the tests, although I'm sure that this will be enough time

Reply:
I think so too, in that case I will include the rework as part of this.

Rest assured that introducing the `ready chan<- struct{}` signal would remove the need for `time.Sleep()`.